macros.go

```diff
@@ -128,7 +128,7 @@ func trimAll(s []string) []string {
 }
 
 func getMacroRegex(name string) string {
-	return fmt.Sprintf("\\$__%s\\b(?:\\((.*?)\\))?", name)
+	return fmt.Sprintf("\\$__%s\\b(?:\\((.*?\\)?)\\))?", name)
```

Comment:
we may need to modify this a bit more, if there is another nested function this won't work (e.g. `$__timeFilter(cast(reduce(sth) as timestamp))`).

I wonder if we should keep making this regex more complex or go with something else like splitting the string but I don't see a good solution for that either.

Reply:
I don't have a good suggestion for a regex here, if we are too flexible it would match several macros with one match and if we are more strict we would only match N number of nested functions (and the regex would be quite complex).

Option B would be to not use a regex and parse the macro letter by letter and finish when the number of `(`s matches the `)`s.

Since that may be complex to write with the current code and since the code is with your current PR better than it was, feel free to not implement that.